fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+def make_synthetic_dataset(feature_config=None, n_features=20, n_informative=4, random_state=None):
+    """Create a synthetic dataset with a single sensitive feature: 'gender'.
+
+    Parameters
+    ----------
+    feature_config : dict, default=None
+        A dict whose keys are the labels for the gender feature, and whose set
+        the :code:`n_samples` and :code:`class_sep` parameters for :code:`make_classification`.
+        If `None`, the following dictionary is used by default::
+
+            {
+                'Man': {'n_samples': 500, 'class_sep': 1},
+                'Other': {'n_samples': 500, 'class_sep': 0.5},
+                'Unspecified': {'n_samples': 500, 'class_sep': 0.5},
+                'Woman': {'n_samples': 500, 'class_sep': 2},
+            }
+
+    n_features : int, default=20
+        The total number of features.
+
+    n_informative : int, default=4
+        The number of informative features.
+
+    random_state : int or RandomState instance, default=None
+        The random number generator seed or :class:`numpy.random.RandomState` to use.
+
+    Returns
+    -------
+    (X, y, gender) : tuple of numpy.ndarray
+        X : numpy.ndarray
+            The generated samples.
+        y : numpy.ndarray
+            Labels for the binary classification.
+        gender : numpy.ndarray
+            The sensitive feature label.
+    """
+    rng = check_random_state(random_state)
+
+    classification_kwargs = {
+        'n_features': n_features,
+        'n_informative': n_informative,
+        'n_classes': 2,
+        'random_state': rng,
+    }
+
+    if feature_config is None:
+        feature_config = {
```

Comment:
do we want these to be gender? Or is it enough to have them as `A, B, C, ...`? It doesn't matter what those categories are, does it?

Reply:
I think it should be configurable (as mentioned in another comment).
You'd be surprised how often I've explained something with A, B, C, D and people didn't get it until I said "well, imagine that these are gender or age groups. I'm fine with it being something generic in the first version, though. Moving to a configurable option in a future PR is also in line with what I wrote in the other comment.